components/map/Clusters.js

```diff
@@ -58,27 +69,64 @@ export default function Clusters({users}) {
 
   return (
     <>
-      {clusters.map(cluster => {
+      {userClusters.map(userCluster => {
           // every cluster point has coordinates
-          const [longitude, latitude] = cluster.geometry.coordinates;
+          const [longitude, latitude] = userCluster.geometry.coordinates;
           // the point may be either a cluster or user
           const {
             cluster: isCluster,
             point_count: pointCount,
             username
-          } = cluster.properties;
+          } = userCluster.properties;
+
+          // we have a cluster to render
+          if (isCluster) {
+            return (
+              <Marker
+                key={`cluster-${userCluster.id}`}
+                position={[latitude, longitude]}
+                icon={fetchIcon(pointCount)}
+                eventHandlers={{
+                  click: () => {
+                    const expansionZoom = Math.min(
+                      userSupercluster.getClusterExpansionZoom(userCluster.id),
+                      18
+                    );
+                    map.setView([latitude, longitude], expansionZoom, {
+                      animate: true
+                    });
+                  }
+                }}
+              />
+            );
+          }
+
+          // we have a single point to render
+          return (
+            <UserMarker user={userCluster} key={username} />
```

Comment:
Use a check on the data here to determine whether we have a user or event.

Reply:
So, after putting a check, if it is a User then I have to put a UserMarker and if it is an event, I have to put EventMarker, correct ?